### DISCUSSION TITLE
Importing package `@thorswap-lib/toolbox-evm` breaks app with vague error message

### DISCUSSION THREAD

**Discussion by NotoriousLisp** [vercel/next.js#45133] [Help]
Upon importing the package `@thorswap-lib/toolbox-evm`, our app throws the error:

`error - Class extends value undefined is not a constructor or null` 
The output also includes a Nextjs Info message of "_This might be caused by a React Class Component being rendered in a Server Component, React Class Components only works in Client Components. Read more: https://nextjs.org/docs/messages/class-component-in-server-component_"

This error appears whether the package is imported into a server side or client side component.
I'm truly lost on what is causing this issue and not even sure how to go about debugging it at this point.  This package works fine in other React applications but not within Nextjs.

**Sample Repo Reproducing Error**
https://github.com/NotoriousLisp/nextjs-thorswap-issue

**Comment by 0xsarwagya** [comment]
The error message "Class extends value undefined is not a constructor or null" and the Next.js info message "This might be caused by a React Class Component being rendered in a Server Component, React Class Components only works in Client Components" suggest that the issue is related to the use of a class component in a server-side component.

It appears that the package @thorswap-lib/toolbox-evm is using class components, which are not supported in server-side components in Next.js. You can try converting the class components to functional components with hooks.

Another possible solution is to use dynamic imports to load the package only on the client-side. This can be done by using the import() syntax and the useEffect hook to dynamically import the package only when the component is mounted on the client-side.

I hope this helps. If you have any other question, please let me know.
